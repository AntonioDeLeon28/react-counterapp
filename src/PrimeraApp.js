import React from 'react';
// import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
 
const PrimeraApp = ( { saludo3 = 'Hola Vegeta', subtitulo } ) => {

    const saludo = {
        nombre: 'Fernado',
        edad: 34
    }
    const saludo2 = 'Hola Antonio';
    return ( // Forma corta de agegar un Fagment
        <> 
            <pre>{ JSON.stringify( saludo, null, 3 ) }</pre>
            <h1>{ saludo2 }</h1> 
            <h1>{ saludo3 }</h1>
            {/* <p>Mi primera aplicación</p> */}
            <p>{ subtitulo }</p>
        </>
    ); 

}

PrimeraApp.propTypes = {
    saludo3: PropTypes.string.isRequired // No importa que pongamos valor por default en la función, va a marcar un error en la consola aunque sí va mostrar en la página web el valor por default
}

PrimeraApp.defaultProps = { // Es otra forma de darle valor por default a los argumentos que recibe una funión
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;

