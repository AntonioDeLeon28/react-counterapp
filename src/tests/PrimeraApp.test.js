import '@testing-library/jest-dom';

import React from 'react';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', () => { 

    test('Debe de mostrar <PrimeraApp /> correctamente', () => { 

        const saludo = 'Hola, Soy Goku';
        const wrapper = shallow( <PrimeraApp saludo3={ saludo }/> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de mostrar el subtitulo enviado por props', () => { 

        const saludo = 'Hola, Soy Goku';
        const subTituloDeProps = 'Soy un subtitulo';
        const wrapper = shallow( <PrimeraApp saludo3={ saludo }/> ); //No le estamos enviando un subtitulo, as√≠ que lo va a sacar de default de props

        const textoParrafo = wrapper.find('p').text(); 
        console.log(textoParrafo);
        expect( textoParrafo ).toBe( subTituloDeProps );

    });

})