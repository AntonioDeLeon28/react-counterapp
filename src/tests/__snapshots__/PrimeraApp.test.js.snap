// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <pre>
    {
   "nombre": "Fernado",
   "edad": 34
}
  </pre>
  <h1>
    Hola Antonio
  </h1>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
